services/user/XAdmin/ui/src/hooks/useKeyDevices.ts

```diff
@@ -0,0 +1,56 @@
+import { useQuery, useMutation } from "@tanstack/react-query";
+
+import { queryKeys } from "@/lib/queryKeys";
+import { chain } from "@/lib/chainEndpoints";
+import { exportKeyToDER, generateP256Key } from "@/lib/keys";
+
+export const useKeyDevices = () =>
+    useQuery({
+        queryKey: queryKeys.keyDevices,
+        queryFn: async () => {
+            try {
+                return chain.getKeyDevices();
+            } catch (e) {
+                console.error("Failed to fetch key devices", e);
+                throw "Failed to fetch key devices";
```

Comment:
For the sake of consistency I think we should aim for all promises / hooks to throw an `Error` rather than a `string` so then when displaying them on the UI it's the same, otherwise we'll end up with more `e instanceof Error` and `typeof e == 'string' to try parse some kinda message to display.
```suggestion
                throw new Error("Failed to fetch key devices");
```